bundles/org.openhab.binding.vigicrues/src/main/java/org/openhab/binding/vigicrues/internal/VigiCruesHandlerFactory.java

```diff
@@ -42,17 +44,23 @@
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.vigicrues")
 @NonNullByDefault
 public class VigiCruesHandlerFactory extends BaseThingHandlerFactory {
-    private final Gson gson;
-    // Needed for converting UTC time to local time
-    private final TimeZoneProvider timeZoneProvider;
+    private final LocationProvider locationProvider;
+    private final ApiHandler apiHandler;
+    private @Nullable ServiceRegistration<?> serviceReg;
 
     @Activate
-    public VigiCruesHandlerFactory(@Reference TimeZoneProvider timeZoneProvider) {
-        this.timeZoneProvider = timeZoneProvider;
-        this.gson = new GsonBuilder()
-                .registerTypeAdapter(ZonedDateTime.class, (JsonDeserializer<ZonedDateTime>) (json, type,
-                        jsonDeserializationContext) -> ZonedDateTime.parse(json.getAsJsonPrimitive().getAsString()))
-                .create();
+    public VigiCruesHandlerFactory(@Reference TimeZoneProvider timeZoneProvider,
+            @Reference LocationProvider locationProvider) {
+        this.locationProvider = locationProvider;
+        this.apiHandler = new ApiHandler(timeZoneProvider);
+    }
+
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        VigiCruesDiscoveryService discoveryService = new VigiCruesDiscoveryService(apiHandler, locationProvider);
+        serviceReg = bundleContext.registerService(DiscoveryService.class.getName(), discoveryService,
+                new Hashtable<>());
```

Comment:
Why isn't the VigiCruesDiscoveryService a `@Component`?

Reply:
Because it needs the ApiHandler

Reply:
Why not make the ApiHandler a `@Component` as well then?

Reply:
Did it.